### ISSUE TITLE
Undefined method `retrieve_work' for `Sidekiq::Pro::BasicFetch:Module`

### ISSUE THREAD

**Issue by dferdian** [mperham/sidekiq#4616]
Ruby version: 2.7.0
Sidekiq / Pro / Enterprise version(s): sidekiq 6.1.0 / sidekiq-pro 5.1.0

Please include your initializer and any error message with the full backtrace.

```
2020-07-01T04:05:25.211Z pid=67119 tid=c7 ERROR: Error fetching job: undefined method `retrieve_work' for Sidekiq::Pro::BasicFetch:Module
2020-07-01T04:05:25.226Z pid=67119 tid=c7 WARN: NoMethodError: undefined method `retrieve_work' for Sidekiq::Pro::BasicFetch:Module
2020-07-01T04:05:25.226Z pid=67119 tid=c7 WARN: /Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/processor.rb:83:in `get_one'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/processor.rb:95:in `fetch'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/processor.rb:77:in `process_one'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/processor.rb:68:in `run'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/util.rb:15:in `watchdog'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/util.rb:24:in `block in safe_thread'
```

And there is another error when I try to stop the worker.

```
2020-07-01T04:05:43.664Z pid=67119 tid=1k3b DEBUG: Got INT signal
2020-07-01T04:05:43.664Z pid=67119 tid=1k3b INFO: Shutting down
2020-07-01T04:05:43.664Z pid=67119 tid=1k3b INFO: Terminating quiet workers
2020-07-01T04:05:43.664Z pid=67119 tid=5xv INFO: Scheduler exiting...
2020-07-01T04:05:43.771Z pid=67119 tid=1k3b INFO: Pausing to allow workers to finish...
2020-07-01T04:05:44.395Z pid=67119 tid=1k3b WARN: NoMethodError: undefined method `bulk_requeue' for Sidekiq::Pro::BasicFetch:Module
2020-07-01T04:05:44.396Z pid=67119 tid=1k3b WARN: /Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/launcher.rb:61:in `stop'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/cli.rb:102:in `rescue in launch'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/cli.rb:93:in `launch'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq/cli.rb:83:in `run'
/Users/username/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/bin/sidekiq:31:in `<top (required)>'
/Users/username/.asdf/installs/ruby/2.7.1/bin/sidekiq:23:in `load'
/Users/username/.asdf/installs/ruby/2.7.1/bin/sidekiq:23:in `<top (required)>'
```

Are you using an old version? No

Have you checked the changelogs to see if your issue has been fixed in a later version?
Yes. I use the latest version for both `sidekiq` and `sidekiq-pro`

**Comment by mperham** [comment]
That’s a bug that will be fixed in Sidekiq Pro 5.1.1. The workaround is to enable super_fetch.

**Comment by IanVaughan** [comment]
I also get this when going from 6.0.7 to 6.1.0 (not pro).
So I do not have access to `super_fetch`

```
Sidekiq.configure_server do |config|
  config.super_fetch!
end
```
Results in 
```
2020-07-01T12:16:04.904Z pid=55265 tid=1ecl WARN: NoMethodError: undefined method `super_fetch!' for Sidekiq:Module
2020-07-01T12:16:04.904Z pid=55265 tid=1ecl WARN: /Users/ianvaughan/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/sidekiq-6.1.0/lib/sidekiq.rb:76:in `configure_server'
.../config/initializers/sidekiq.rb:7:in `<main>'
```

**Comment by mperham** [comment]
Pro 5.1.1 is out.

**Comment by Adithya-copart** [comment]
> That’s a bug that will be fixed in Sidekiq Pro 5.1.1. The workaround is to enable super_fetch.

@mperham Upgrading to `Sidekiq Pro 5.1.1.` fixed this for me. However, I ran into this issue in `5.0.1` with super_fetch enabled as well.

```
2020-07-20T18:38:28.751Z pid=7315 tid=5et INFO: Running in jruby 9.2.9.0 (2.5.7) 2019-10-30 458ad3e OpenJDK 64-Bit Server VM 25.242-b08 on 1.8.0_242-b08 +jit [darwin-x86_64]
2020-07-20T18:38:28.751Z pid=7315 tid=5et INFO: Sidekiq Pro 5.0.1, commercially licensed.  Thanks for your support!
2020-07-20T18:38:29.460Z pid=7315 tid=5et INFO: Starting processing, hit Ctrl-C to stop
2020-07-20T18:38:29.460Z pid=7315 tid=429 INFO: Sidekiq::Pro::SuperFetch
2020-07-20T18:38:29.581Z pid=7315 tid=42f ERROR: Error fetching job: undefined method `retrieve_work' for Sidekiq::Pro::SuperFetch:Class
```
